File: 206. Reverse Linked List.go
package main

import (
	"fmt"
)

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseList(head *ListNode) *ListNode {
	cur := head
	var prev *ListNode = nil
	var next *ListNode = nil
	for cur != nil {
		next = cur.Next
		cur.Next = prev
		prev = cur
		cur = next
	}
	head = prev
	return head
}

func main() {
	fmt.Println("Redis Cache")
}

File: 206. Reverse Linked List.swift
public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init() { self.val = 0; self.next = nil; }
    public init(_ val: Int) { self.val = val; self.next = nil; }
    public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
}


class Solution {

    func reverseList(_ head: ListNode?) -> ListNode? {
        
        var prev: ListNode? = nil
        var next: ListNode? = nil
        var cur = head
        
        while cur != nil {
            next = cur!.next
            cur!.next = prev
            prev = cur
            cur = next
        }
            
        return prev
    }
}